spec/requests/users_spec.rb

```diff
@@ -71,12 +92,49 @@
   end
 
   it 'should return 400 if user does not exist' do
-    expected_response = { 'error' => 'User Not Found' }
-
     expect do
       delete '/users/420'
     end.to_not change { User.count }
-    expect(json_body).to eq(expected_response)
+    expect(response.status).to eq(400)
+  end
+end
+
+RSpec.describe 'POST /users/:id/favorites' do
+  it 'should create a favorite for corresponding park and user' do
+    park = create(:skatepark)
+    user = create(:user)
+
+    expect do
+      post "/users/#{user.id}/favorites", park_id: park.id
+    end.to change { user.skateparks.count }.by(1)
+
+    fav = Favorite.where(user_id: user.id, skatepark_id: park.id)
+    expect(fav.first).to be_truthy
+  end
+
+  it 'should return 400 if favorite already exists' do
+    park = create(:skatepark)
+    user = create(:user)
+    user.skateparks << park
+
+    expect do
+      post "/users/#{user.id}/favorites", park_id: park.id
+    end.to_not change { user.skateparks.count }
+    expect(response.status).to eq(400)
+  end
+
+  it 'should return 400 if skatepark or user does not exist' do
+    park = create(:skatepark)
+    user = create(:user)
+
+    expect do
+      post "/users/#{user.id}/favorites", park_id: 5
+    end.to_not change { user.skateparks.count }
+    expect(response.status).to eq(400)
+
+    expect do
+      post '/users/1/favorites', park_id: park.id
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.